components/Base/Modal.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <TransitionRoot
+    appear
+    :show="isShown"
+    as="template"
+  >
+    <Dialog
+      as="div"
+      class="relative z-10"
+      @click="$emit('close')"
+    >
+      <TransitionChild
+        as="template"
+        enter="duration-300 ease-out"
+        enter-from="opacity-0"
+        enter-to="opacity-100"
+        leave="duration-200 ease-in"
+        leave-from="opacity-100"
+        leave-to="opacity-0"
+      >
+        <div class="fixed inset-0 bg-black bg-opacity-25" />
+      </TransitionChild>
+
+      <div class="fixed inset-0 overflow-y-auto">
+        <div
+          class="flex min-h-full items-center justify-center p-4 text-center"
+        >
+          <TransitionChild
+            as="template"
+            enter="duration-300 ease-out"
+            enter-from="opacity-0 scale-95"
+            enter-to="opacity-100 scale-100"
+            leave="duration-200 ease-in"
+            leave-from="opacity-100 scale-100"
+            leave-to="opacity-0 scale-95"
+          >
+            <DialogPanel
+              v-loading="isLoading"
+              class="w-full relative max-w-md transform overflow-hidden rounded-2xl bg-white p-5 text-left align-middle shadow-xl transition-all"
+            >
+              <button
```

Comment:
`BaseButton` ?

Reply:
We don't have a variant for it. It's a really small close button. 

<img width="200" alt="image" src="https://user-images.githubusercontent.com/66744401/224112126-38dc01d0-005a-4fc1-81e5-31ffc5a0c3ec.png">
<img width="300" alt="image" src="https://user-images.githubusercontent.com/66744401/224112183-3dba77dc-11e5-44c6-ae0a-f3d6588d70f7.png">


I also saw you using a new component for counter buttons in your PR. So maybe we can tweak the BaseButton with one more size variant, or create something like `IconButton` component?

Reply:
I've added a variant to BaseButton - `variant="content"`. This variant basically means that button has no styling and it's just a wrapper for its **content**, like in this case, where we have an svg icon inside the button, and it's purpose is to be the button's look 😄

Reply:
@zakrzaq I double checked the behaviour and it turned out `variant="text"` worked exactly the same way there so I don't think this `variant="content"` is anyhow beneficial. I removed it and just used `text` 😅